interview-questions.md

```diff
@@ -8,51 +8,65 @@ Then, research the question to expand on your answer. Even if you feel you have
 
   Researched answer:
-
+  Props is a special keyword in React that stands for properties. Props is used to pass data and methods between components. Information gets passed from the parent component to the child component through the component call. Props come from a component that holds state and are a snapshot of that components state. Props are passed onto components that will display the information or will let the user interact with the information. 
+  
 
 
 2. What is a DOM event?
 
   Your answer:
+  DOM events recognize interactions with a user and respond depending on the type of event. A DOM event is an object that looks at key:value pairs. The event is an object that is provided by an onChange or onClick attribute. The document object model listens for any changes made to a certain node and will take action.
 
   Researched answer:
+  The DOM is listening for a change on a particular node using an event listener and will execute a corresponding action. The Document Object Model listens for changes and in order for web applications to recognize user interactions and respond to a particular event listeners are added to HTML (and JSX) tags. The event listeners wait for certain events to happen. Two event listeners that can be used are onClick and onChange. The onClick attribute can be added to a button and the onChange attribute is useful when users are typing in text fields.
 
 
 
 3. What is object-oriented programming? How is it different than functional programming?
 
   Your answer:
+  Object-oriented programming is a way of creating containers for data that could be changed and manipulated without affecting the entire program. Object-oriented programming (OOP) is a programming model that focuses on the objects that need to be manipulated instead of the logic required to manipulate them. The data is the information stored in variables and the behavior are the methods that belong to the object. In object-oriented programsming objects and methods are the key elements and in functional programming the variables and functions are the main elements of the code.
 
   Researched answer:
+  Object-oriented programming (OOP) is a computer programming model that organizes software design around data, or objects, rather than functions and logic. An object can be defined as a data field that has unique attributes and behavior. OOP focuses on the objects that developers want to manipulate rather than the logic required to manipulate them. Additional benefits of OOP include code reusability, scalability and efficiency. Object-oriented programming uses statements that change a programs state. The programming model used in functional programming expresses the logic and does not change state. Functions in functional programming do not modify variables, but they can generate new variables as an output. 
```

Comment:
[Hey I've seen this article be used and sliced up before ](https://www.techtarget.com/searchapparchitecture/definition/object-oriented-programming-OOP)

Thats a bummer.